src/dagnode-stream.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const TransformStream = require('readable-stream').Transform
+const streamToValue = require('./stream-to-value')
+const getDagNode = require('./get-dagnode')
+
+/*
+  Transforms a stream of objects to DAGNodes and outputs them as objects.
+
+  Usage: inputStream.pipe(DAGNodeStream({ send: send }))
+
+  Input object format:
+  {
+    Name: '/path/to/file/foo.txt',
+    Hash: 'Qma4hjFTnCasJ8PVp3mZbZK5g2vGDT4LByLJ7m8ciyRFZP'
+  }
+
+  Output object format:
+  {
+    path: '/path/to/file/foo.txt',
+    hash: 'Qma4hjFTnCasJ8PVp3mZbZK5g2vGDT4LByLJ7m8ciyRFZP',
+    size: 20
+  }
+*/
+class DAGNodeStream extends TransformStream {
+  constructor (options) {
+    const opts = Object.assign(options || {}, { objectMode: true })
+    super(opts)
+    this._send = opts.send
+  }
+
+  static streamToValue (send, inputStream, callback) {
```

Comment:
Is it? Is it not? And if so, why is that important. Provide context and solutions or guidance.